components/session-based-conditional-auth-functions/src/main/java/org/wso2/carbon/identity/extensions/authenticator/conditional/auth/functions/function/ExecuteActionFunction.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+package org.wso2.carbon.identity.extensions.authenticator.conditional.auth.functions.function;
+
+import org.wso2.carbon.identity.application.authentication.framework.config.model.graph.js.JsAuthenticationContext;
+
+import java.util.Map;
+
+/**
+ * Function definition for verify some defined requirement is fulfilled.
+ */
+@FunctionalInterface
+public interface ExecuteActionFunction {
+
+    Boolean execute(JsAuthenticationContext context, Map<String, String> map);
```

Comment:
Can we please add method level comment? Please correct in other places as well

Reply:
method level comments added in all missing places. Please check.